src/olympia/files/models.py

```diff
@@ -218,17 +230,19 @@ def latest_xpi_url(self, attachment=False):
 
     @property
     def file_path(self):
-        return os.path.join(
-            user_media_path('addons'), str(self.version.addon_id), self.filename
-        )
+        return self.file.path if self.file else ''
+
+    @property
+    def filename(self):
+        return self.file.name if self.file else ''
 
     @property
     def addon(self):
         return self.version.addon
 
     @property
     def extension(self):
-        return os.path.splitext(self.filename)[-1]
+        return os.path.splitext(self.file.name)[-1]
```

Comment:
This function doesn't have a `... if self.file ...` guard.

Reply:
Looks like it's only really used once in `latest_xpi_url()`, which itself is only used in tests. We have wanted to expose it (https://github.com/mozilla/addons-server/issues/7187) so we should probably keep it... I'll add something

Reply:
3335701804151459d7dbd13637d536b38a985b91